src/main/java/kostka/moviecatalog/service/EsMovieService.java

```diff
@@ -48,14 +49,16 @@ public EsMovie saveMovie(final EsMovie esMovie) {
     }
 
     public List<EsMovie> fullTextSearch(final String term) {
-        LOGGER.info("Searching in ElasticSearch for term '{}' in fields {}, {}, {}",
-                term, NAME_FIELD, DESCRIPTION_FIELD, DIRECTOR_FIELD);
         QueryStringQueryBuilder builder = QueryBuilders
                 .queryStringQuery(term)
-                .field(NAME_FIELD)
-                .field(DESCRIPTION_FIELD)
-                .field(DIRECTOR_FIELD)
                 .autoGenerateSynonymsPhraseQuery(true);
+        // get all EsMovie fields which have String type and add its names to query builder using java Reflection
+        Set<Field> esMovieFields = Arrays.stream(EsMovie.class.getDeclaredFields())
```

Comment:
OPTIONAL: Kdyz prejmenujes `esMovieFields ` na `esStringMovieFields `, tak ti to v podstate vysvetluje ten komentar ;)

Reply:
to dava smysl, diky (y)